### ISSUE TITLE
Airflow integration - explore hooking via airflow plugins

### ISSUE THREAD

**Issue by mobuchowski** [MarquezProject/marquez#1198]
Right now, we hook to airflow by subclassing DAG and asking users to use our subclass instead of `airflow.models.DAG`.
Explore hooking our changes to DAG via plugin mechanism: https://airflow.apache.org/docs/apache-airflow/stable/plugins.html

**Comment by collado-mike** [comment]
If we can figure this out, I'm strongly in favor! A plugin model is, IMO, almost always better than inheritance. It also gives users the freedom to use their own DAG subclasses without compromising their ability to publish lineage.

**Comment by mobuchowski** [comment]
Well, the plugin model isn't as clean as I would like and would still require some level of hooking up to the internal components. Hovewer, there are some plugins doing similar stuff, so probably it's just a matter of finding the best way to do monkeypatching.
https://github.com/newrelic/newrelic-airflow-plugin
https://github.com/getsentry/airflow-metrics

**Comment by wslulciuc** [comment]
Agreed, using a plugin would be a more ideal approach. It's something we considered initially, but defaulted to extend the DAG class as it _seemed_ a bit more straightforward. Would be great to have a more pluggable model, so +1 on exploring this.
